File: Desarrollo Software-DS/Practica 1/Sesion3/Obligatoria/S3/tecnico.cpp
#include<tecnico.h>

using namespace std;

Tecnico::Tecnico(string n,int e)
{
    this->nombre=n;
    this->edad=e; 
    trbRealizar=nullptr;  
}

void Tecnico::cogerTrabajo(int id,vector  <Trabajo *> trbs)
{
    for(unsigned int i=0; i<trbs.size();i++)
    {
        if(trbs[i]->getID()==id && !trbs[i]->getOcupado())
        {
            trbRealizar=trbs[i];
            trbs[i]->setOcupado(true);
            cout<<"El tecnico "<<this->nombre<<" ha obtenido el trabajo con id: "<<id<<endl;
        }
    }
}


void Tecnico::realizarTrabajo()
{
    cout<<"El tecnico " <<this->nombre <<" realiza el trabajo con id: "<<this->trbRealizar->getID()<<endl;
}

void Tecnico::cobrar()
{
    cout<<"El tecnico " <<this->nombre<<" ha cobrado el trabajo con id "<<this->trbRealizar->getID() <<" con una cantidad de "<<this->trbRealizar->getPresupuesto()<<endl;
}

File: Tercero/Desarrollo Software-DS/Practica 1/Sesion3/Opcional/dart/lib/tecnico.dart
import 'trabajo.dart';


class Tecnico {
  String nombre = "";
  int edad = 0;
  Trabajo trbRealizar = new Trabajo(0, "", 0);

  Tecnico(String n, int e) {
    this.nombre = n;
    this.edad = e;
  }

  String getNombre(){return nombre;}
  void setNombre(String n){nombre=n;}
  int getEdad(){return edad;}
  void setEdad(int e){edad=e;}
  Trabajo getTrabajo(){return trbRealizar;}

  void cogerTrabajo(int id, List<Trabajo> trbs) {
    for (var i = 0; i < trbs.length; i++) {
      if (trbs[i].getID() == id && !trbs[i].getOcupado()) {
        trbRealizar = trbs[i];
        trbs[i].setOcupado(true);
        print("El tecnico "+this.nombre+" ha obtenido el trabajo con id: "+id.toString()+"\n");
      }
    }
  }

void realizarTrabajo()
{
    print("El tecnico "+ this.nombre +" realiza el trabajo con id: "+this.trbRealizar.getID().toString()+"\n");
}


void cobrar()
{
    print("El tecnico " +this.nombre+" ha cobrado el trabajo con id "+this.trbRealizar.getID().toString() +" con una cantidad de "+this.trbRealizar.getPresupuesto().toString());
}

}


